src/main/java/lotto/LottoNumber.java

```diff
@@ -9,35 +10,42 @@
 public class LottoNumber implements Comparable<LottoNumber> {
 
 	public static final int LOTTO_MIN_NUMBER = 1;
-	public static final int LOTTO_MAX_NUMBER = 45;
+	public static final int LOTTO_LIMIT_NUMBER = 45 + 1;
 
-	private static final Map<Integer, LottoNumber> NUMBER_CACHE
-		= IntStream.range(LOTTO_MIN_NUMBER, LOTTO_MAX_NUMBER + 1)
+	private static final Map<Integer, LottoNumber> NUMBER_CACHE;
+
+	static {
+		NUMBER_CACHE = IntStream.range(LOTTO_MIN_NUMBER, LOTTO_LIMIT_NUMBER)
 			.boxed()
 			.collect(toMap(i -> i, LottoNumber::new));
+	}
 
 	private final int value;
 
-	public static LottoNumber valueOf(int value) {
-		if (NUMBER_CACHE.containsKey(value)) {
-			return NUMBER_CACHE.get(value);
-		}
-
-		return NUMBER_CACHE.put(value, new LottoNumber(value));
-	}
-
 	private LottoNumber(int value) {
 		validateNumberValue(value);
 
 		this.value = value;
 	}
 
 	private static void validateNumberValue(int value) {
-		if (value < LOTTO_MIN_NUMBER || LOTTO_MAX_NUMBER < value) {
+		if (value < LOTTO_MIN_NUMBER || LOTTO_LIMIT_NUMBER <= value) {
 			throw new IllegalArgumentException("Number must be between 1 and 45");
 		}
 	}
 
+	public static LottoNumber valueOf(int value) {
+		if (NUMBER_CACHE.containsKey(value)) {
+			return NUMBER_CACHE.get(value);
+		}
+
+		return NUMBER_CACHE.put(value, new LottoNumber(value));
+	}
+
+	public static LottoNumber valueOf(String value) {
+		return valueOf(parseInt(value));
+	}
```

Comment:
```suggestion
	public static LottoNumber valueOf(String value) {
		return valueOf(parseInt(value.trim()));
	}
```
이렇게 해주면 mapping 할 때 조금 더 직관적으로 만들 수 있지 않을까요?